### PR TITLE
Adding support for domain resource; bumping go-enf version

### LINKED ISSUE #15: Add Domain resource support
Create a new resource for domain and integrate with the terrraform provider.

### PR DESCRIPTION
**Pull request by dylanirlbeck**
Fixes #15 

This PR should **not** be merged in until Venkat addresses [this issue](https://github.com/xaptum/xcr/issues/66) with xcr. I just wanted some early feedback since that issue should be fixed relatively soon.

Essentially, we cannot `GET` an individual domain by its /48 address, and thus cannot perform the `resourceXaptumDomainRead` function. I chose to use the /48 address as the resource ID, rather than the id returned by xcr, for readability's sake.

### CODE REVIEW

File: xaptum/provider.go [hunk 1]
```diff
@@ -71,3 +72,20 @@ func validateRateLimitRequest(d *schema.ResourceData) error {
...
+	nameHasChanged := d.HasChange("name")
+	typeHasChanged := d.HasChange("type")
+	adminEmailHasChanged := d.HasChange("admin_email")
+	adminNameHasChanged := d.HasChange("admin_name")
+	if nameHasChanged || typeHasChanged || adminEmailHasChanged || adminNameHasChanged {
+		return errors.New("Can only update status. name, type, admin_email, and admin_name cannot change")
+	}
+
+	status := d.Get("status").(string)
+	if status != "READY" || status != "ACTIVE" {
```
Comment (drbild) [inline xaptum/provider.go]: `||` should be `&&`.

File: xaptum/resource_xaptum_domain.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	if err != nil {
+		return err
+	}
+
+	client := meta.(*EnfClient).Client
+	address := d.Get("network").(string)
+
+	switch d.Get("status") {
+	case "READY":
+		client.Domains.DeactivateDomain(context.Background(), address)
```
Comment (drbild) [inline xaptum/resource_xaptum_domain.go]: Need to check the return values of `DeactiveDomain` and `ActivateDomain` for errors.

Comment (dylanirlbeck) [reply xaptum/resource_xaptum_domain.go]: Fixed in latest commit.

**Comment by dylanirlbeck** [comment]
@drbild I think it's about time we bumped the version of `terraform-provider-xaptum`. What do you think?

**Comment by drbild** [comment]
> @drbild I think it's about time we bumped the version of `terraform-provider-xaptum`. What do you think?

I agree. After this gets merged, you can cut a new release.

**Pull request merged**